messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -131,6 +133,35 @@ protected void initializeConfig(ClientConfig config) throws IOException {
         + " streamType:" + streamType);
   }
 
+  private void startNewCollectorDiscovererTimer(ClientConfig config) {
+       /*
+    This timer is responsible for identifying new
+    collector output sub directories and initialising partition readers for them
+     */
+     int discoverFrequency = config.getInteger(frequencyForDiscoverer,
+        DEFAULT_FREQUENCY_FOR_DISCOVERER);
+    partitionDiscovererAndReaderCreator = new Timer(
```

Comment:
timer task is recreated and rescheduled in the doreset method. I will extend the current test case to have a reset as well. and I will dynamically add one more collector to cover this case